Fusion content (#7794)
fishtown-analytics/docs.getdbt.com: website/snippets/_install-dbt-extension.md

> The dbt extensions for VS Code and Cursor streamlines dbt development workflows. The dbt extension is powered by the <constant name="fusion_engine" />.

Reviewer: Constant not displayed

> The dbt extensions for VS Code and Cursor streamlines dbt development workflows. The dbt extension is powered by the <constant name="fusion_engine" />.

Reviewer: ```suggestion
The dbt extension for VS Code and Cursor streamlines dbt development workflows. The dbt extension is powered by the <constant name="fusion_engine" />.
```

> The dbt extensions for VS Code and Cursor streamlines dbt development workflows. The dbt extension is powered by the <constant name="fusion_engine" />.

Reviewer: ```suggestion
The dbt extensions for VS Code and Cursor streamlines dbt development workflows. The dbt extension is powered by the <Constant name="fusion_engine" />.
```

> The dbt extensions for VS Code and Cursor streamlines dbt development workflows. The dbt extension is powered by the <Constant name="fusion_engine" />.

Reviewer: ```suggestion
The dbt extension for VS Code and Cursor streamlines dbt development workflows. The dbt extension is powered by the <Constant name="fusion_engine" />.
```

## Prerequisites

To use the extension, you must meet the following prerequisites:

> - The dbt extension requires installation of the <constant name="fusion_engine" />. <constant name="fusion" /> installation is part of the extension installation process, but you can also [manually install](/docs/fusion/install-fusion) separate from this workflow, either before or after the extension is installed. 

Reviewer: Constants not displayed

- You are using the [VS Code](https://code.visualstudio.com/) or [Cursor](https://www.cursor.com/en) code editor.

> - You are not using (or have disabled) 3rd party dbt extensions.

Reviewer: ```suggestion
- You are not using (or have disabled) third-party dbt extensions.
```

    <Lightbox src="/img/docs/extension/install-dbt-fusion-engine.png" width="60%" title="Follow the prompt to install the dbt Fusion engine"/>
5. You're all set up! See [about the dbt extension](/docs/about-dbt-extension) for more information on how to use the dbt extension.
    <Lightbox src="/img/docs/extension/kitchen-sink.png" width="60%" title="Showing lineage and compiled code in the extension"/>

## Upgrade to Fusion

:::note

> If you are already running the <constant name="fusion_engine" />, you must be on version `2.0.0-beta.66` or higher to use the migration tool.

Reviewer: Constant not displayed

:::

> The dbt extension provides a built-in migration tool to walk you through the process of configuring <constant name="fusion" />  and updating your dbt project to support all of its features and fix any deprecated code. To start the process:

Reviewer: Constant not displayed

1. From the VS Code left-side menu, click the **dbt logo**.
2. In the resulting pane, open the **Get started** section and click the **Get started** button. 

    <Lightbox src="/img/docs/extension/fusion-onboarding-experience.png" title="The dbt extension help pane and migration assistant." width="60%" /> 

You can also manually start this process by opening a CLI window and running: 

```
dbt init --fusion-upgrade
```

This will start the migration tool and guide you through the migration with a series of prompts:

> - **Do you have an existing dbt platform account**: If you answer `Y`, you will be given instructions for downloading your dbt platform profile to register the extension. An `N` answer will skip to the next step.

Reviewer: ```suggestion
- **Do you have an existing dbt platform account?**: If you answer `Y`, you will be given instructions for downloading your dbt platform profile to register the extension. An `N` answer will skip to the next step.
```

- **Ready to run a dbtf init?** (If there is no `profiles.yml` file present): You will go through the dbt configuration processes, including connecting to your data warehouse. 
- **Ready to run a dbtf debug?** (If there is an existing `profiles.yml` file): Validates that your project is configured correctly and can connect to your data warehouse.

> - **Ready to run a dbtf parse?**: Your dbt project will be parsed to check for compatibility with <constant name="fusion" />.

Reviewer: Constant not displayed

>     - If any issues are encountered during the parsing, you'll be given the option to run the [dbt-autofix](https://github.com/dbt-labs/dbt-autofix?tab=readme-ov-file#installation) tool to resolve the errors. If you opt not to run the tool during the migration processes, you can always run it later or manually fix any errors. However, the migration tool can not continue until the errors are resolved.

Reviewer: ```suggestion
    - If any issues are encountered during the parsing, you'll be given the option to run the [dbt-autofix](https://github.com/dbt-labs/dbt-autofix?tab=readme-ov-file#installation) tool to resolve the errors. If you opt to not run the tool during the migration processes, you can always run it later or manually fix any errors. However, the migration tool cannot continue until the errors are resolved.
```

> - **Ready to run a ‘dbtf compile -static-analysis off’?** (Only runs once the parse passes): Compiles your project without any static analysis, mimicking dbt Core. This compile only renders Jinja into SQL, so <constant name="fusion" />'s advanced SQL comprehension is temporarily disabled. 

Reviewer: Constant not displayed

- **Ready to run a ‘dbtf compile’?**: Compiles your project with full <constant name="fusion" /> static analysis. It checks that your SQL code is valid in the context of your warehouse's tables and columns. 

    <Lightbox src="/img/docs/extension/fusion-onboarding-complete.png" title="The message received when you have completed migrating your project to the dbt Fusion engine." width="60%" /> 

> Once the migration tool is completed, you're ready to dive into all the features that the <constant name="fusion_engine" /> has to offer!

Reviewer: Constant not displayed

- Users with an existing dbt account can connect their account using a `dbt_cloud.yml` credentials file.

The VS Code extension is free for organizations for up to 15 users.

### New user registration

If you do not already have a dbt account, you'll need to get registered. This only takes a minute!
1. Click the registration prompt in your editor.
     <Lightbox src="/img/docs/extension/registration-prompt.png" width="60%" title="The extension registration prompt in VS Code."/>
2. Accept any prompts to open the link in your browser.
3. Fill out the registration form, then click **Continue**.
    <Lightbox src="/img/docs/extension/registration-screen.png" width="60%" title="The extension registration page in the browser."/>
4. You will receive an email with a verification link. Once you click it, your registration is complete!

### Accessing your dbt account

Registering to use the dbt extension makes it easy to create a full dbt account. You can follow these

> steps to finish setting up your account (_note: this is not required to use the dbt extension_).

Reviewer: ```suggestion
steps to finish setting up your account (_Note: this is not required to use the dbt extension_).
```

1. Navigate to [us1.dbt.com](https://us1.dbt.com) and click **Forgot password?**.
2. Enter the email address you used for your dbt extension registration and click **Continue**.
3. Check your email for a verification link and follow the password reset instructions to set a password for your account.

> Now that you have activated your dbt developer account, you can access features of the dbt platform. You can also re-download your registration key using the steps outlined in [Register with an existing dbt account](#register-with-an-existing-dbt-account) below if you need to set up the dbt extension on a new machine.

Reviewer: ```suggestion
Now that you have activated your dbt developer account, you can access features of the <Constant name="dbt_platform" />. You can also re-download your registration key using the steps outlined in [Register with an existing dbt account](#register-with-an-existing-dbt-account) if you need to set up the dbt extension on a new machine.
```

### Register with an existing dbt account 

<!-- This anchor is linked from the VS Code registration page. Please do not change it -->

If you already have a dbt account, you do not need to re-register to use the dbt extension. The dbt extension can authenticate with the dbt platform using a `dbt_cloud.yml` file. If this file is present in your `~/.dbt/` folder, then the registration flow will automatically attempt to use this

> file during registration. If you do not have a `~/.dbt/dbt_cloud.yml` file downloaded, follow the directions below:

Reviewer: ```suggestion
file during registration. If you do not have a `~/.dbt/dbt_cloud.yml` file downloaded, refer to the following instructions:
```

1. Log in to your dbt account.
2. Click your account name at the bottom of the left-side menu and click **Account settings**.
3. Under the **Your profile** section, click **VS Code Extension**. 
4. In the **Set up your credentials** section, click **Download credentials**. This downloads the `dbt_cloud.yml` file. 
    <Lightbox src="/img/docs/extension/download-registration-2.png" width="60%" title="Download the dbt_cloud.yml file to complete registration."/>
5. Move the downloaded `dbt_cloud.yml` file to your `~/.dbt/` directory.
6. To update your registration in VS Code, open the command palette (`ctrl+shift+P` (<!--Windows/-->Linux) or `cmd+shift+p` (macOS)), then select `dbt: Register dbt extension` to complete the registration.

</Expandable>

<Expandable alt_header="For dbt accounts without Fusion enabled">

1. Log in to your dbt account.
2. Click your account name at the bottom of the left-side menu and click **Account settings**.
3. Under the **Your profile** section, click **CLI**. 
4. In the **Configure Cloud authentication** section, click **Download CLI configuration file**. This downloads the `dbt_cloud.yml` file. 

>     <Lightbox src="/img/docs/extension/download-registration.png" width="60%" title="Download the dbt_cloud.yml file to complete registration."/>

Reviewer: Screenshot not updated

5. Move the downloaded `dbt_cloud.yml` file to your `~/.dbt/` directory.
6. To update your registration in VS Code, open the command palette (`ctrl+shift+P` (<!--Windows/-->Linux) or `cmd+shift+p` (macOS)), then select `dbt: Register dbt extension` to complete the registration.

</Expandable>

## Troubleshooting
<!-- This anchor is linked from the  VS Code extension. Please do not change it -->

#### General troubleshooting tips

> If the dbt extension has activated successfully, you will see the `dbt Extension` label in the status bar at the bottom left of your editor. You can view diagnostic information about the dbt extension by clicking on the **dbt Extension** button.

Reviewer: ```suggestion
If the dbt extension has activated successfully, you will see the `dbt Extension` label in the status bar at the bottom left of your editor. You can view diagnostic information about the dbt extension by clicking the **dbt Extension** button.
```

If the dbt extension label is not present, then it is likely that the dbt extension was not installed successfully. If this happens, try uninstalling the extension, restarting your editor, and then reinstalling the extension.

Note: It is possible to "hide" status bar items in VS Code. Double-check if the **dbt Extension** status bar label is hidden by right-clicking on the status bar in your editor. If you see **dbt Extension** in the right-click menu, then the extension has installed successfully.

#### Missing dbt LSP features

> If you are not seeing dbt LSP features in your editor, first consult the general troubleshooting steps above. If you have confirmed that the dbt extension is installed correctly, but you still do not see dbt Language Server features (autocomplete, go-to-definition, hover text, etc):

Reviewer: ```suggestion
If you are not seeing dbt LSP features in your editor, first consult the general troubleshooting steps above. If you have confirmed that the dbt extension is installed correctly, but you still do not see dbt Language Server features (for example, autocomplete, go-to-definition, hover text):
```
